crates/bevy_audio/src/volume_level.rs

```diff
@@ -0,0 +1,165 @@
+/// Represents a volume level value.
+#[derive(Clone, Copy, Debug)]
+pub enum VolumeLevel {
+    /// Volume Amplitude Ratio (unitless)
+    ///
+    /// # Notes
+    ///
+    /// `Self::Amplitude(a)` is only defined for `a` between `f32::MIN_POSITIVE` and `f32::MAX`.
+    /// If `a` is any other value, `VolumeLevel` will be in an invalid state.
+    Amplitude(f32),
+    /// Decibels of Amplitude Ratio (dB)
+    ///
+    /// # Notes
+    ///
+    /// `Self::Decibels(a)` is defined for all non-`NaN` values of `a`. However, values with a
+    /// magnitude larger than `1,000` are largely meaningless due to the logarithmic nature of
+    /// the decibel scale.
+    Decibels(f32),
+}
+
+impl VolumeLevel {
+    /// The amplitude ratio this represents. This value is bounded between
+    /// `f32::MIN_POSITIVE` and `f32::MAX`, with `1.` representing a neutral level.
+    pub fn amplitude(&self) -> f32 {
+        let amplitude = match self {
+            Self::Amplitude(amplitude) => *amplitude,
+            Self::Decibels(db) => db_to_a(*db),
+        };
+
+        debug_assert!(
+            (f32::MIN_POSITIVE..=f32::MAX).contains(&amplitude),
```

Comment:
`0.0` seems like a perfectly valid amplitude to me.

Reply:
You've given me an idea, also resolved!